arm-servicebus/2014-09-01/swagger/servicebus.json

```diff
@@ -0,0 +1,974 @@
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/AuthorizationRules/{authorizationRuleName}/regenerateKeys": {
+      "post": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_RegenerateKeys",
+        "description": "Gets the Primary and Secondary ConnectionStrings to the namespace ",
+        "externalDocs": {
+          "url": "https://msdn.microsoft.com/en-us/library/azure/mt718977.aspx"
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "namespaceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The namespace name."
+          },
+          {
+            "name": "authorizationRuleName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The connection string of the namespace for the specified authorizationRule."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateKeysParameters"
+            },
+            "description": "Parameters supplied to regenerate Auth Rule."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful",
+            "schema": {
+              "$ref": "#/definitions/ResourceListKeys"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CheckAvailabilityParameters": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Gets or sets name"
+        },
+        "location": {
+          "type": "string",
+          "description": "Gets or sets location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets tags."
+        },
+        "isAvailiable": {
+          "type": "boolean",
+          "description": "Gets or sets true if the name is available and can be used to create new Namespace/ServiceBus. Otherwise false."
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "Parameters supplied to the Check Name Availability for Namespace and ServiceBus."
+    },
+    "CheckAvailabilityResource": {
+      "properties": {
+        "id": {
```

Comment:
id, name, location, type, tags are all properties of the base resource in ARM. You can create a definition for REsource and all your Resources can inherit from it. Thus reducing duplication.
Take a look at the storage spec as an example: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json#L945
Also take a look at the x-ms-azure-resource extension. https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-azure-resource.
When this extension is applied, the generated Resource class will implement the IResource empty interface present in the client runtime. Thus making it easy for users to share resources between RPs. At any given time a customer will be dealing with 2-3 RPs in his script.